lit_llama/model.py

```diff
@@ -28,17 +28,15 @@ def build_rope_cache(seq_len: int, n_elem: int, dtype: torch.dtype, base: int =
     # Calculate the product of position index and $\theta_i$
     idx_theta = torch.outer(seq_idx, theta)
 
-    # Cache them
-    cache = torch.polar(torch.ones_like(idx_theta), idx_theta)  # complex64
```

Comment:
DDP doesn't support broadcasting complex64 buffers. 

```
    _sync_module_states(
  File "/home/adrian/.conda/envs/lit-llama/lib/python3.10/site-packages/torch/distributed/utils.py", line 142, in _sync_module_states
    _sync_params_and_buffers(
  File "/home/adrian/.conda/envs/lit-llama/lib/python3.10/site-packages/torch/distributed/utils.py", line 160, in _sync_params_and_buffers
    dist._broadcast_coalesced(
RuntimeError: Input tensor data type is not supported for NCCL process group: ComplexFloat
```

Solution: We can move the computation of polar() to the forward.

Reply:
Yes makes sense. We could even move computing the whole rope cache to forward.

Reply:
Which would also fix the failing test